sentry-android-core/src/main/java/io/sentry/android/core/internal/util/AndroidThreadChecker.java

```diff
@@ -1,22 +1,28 @@
 package io.sentry.android.core.internal.util;
 
+import android.os.Handler;
 import android.os.Looper;
+import android.os.Process;
 import io.sentry.protocol.SentryThread;
-import io.sentry.util.thread.IMainThreadChecker;
+import io.sentry.util.thread.IThreadChecker;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 
 /** Class that checks if a given thread is the Android Main/UI thread */
 @ApiStatus.Internal
-public final class AndroidMainThreadChecker implements IMainThreadChecker {
+public final class AndroidThreadChecker implements IThreadChecker {
 
-  private static final AndroidMainThreadChecker instance = new AndroidMainThreadChecker();
+  private static final AndroidThreadChecker instance = new AndroidThreadChecker();
+  public static long mainThreadId = Process.myTid();
```

Comment:
I think it should be volatile?

Reply:
also, maybe we should name it better, like `mainThreadKernelId` or `mainThreadSystemId` to distinguish from the thread id